contracts/EverlongStrategy.sol

```diff
@@ -0,0 +1,482 @@
+        _tendConfig.minOutput = _minOutput;
+    }
+
+    /// @notice Sets the minimum vault share price when opening a long.
+    /// @param _minVaultSharePrice Minimum vault share price when opening a
+    ///        long.
+    function setMinVaultSharePrice(
+        uint256 _minVaultSharePrice
+    ) external onlyKeepers {
+        _tendConfig.minVaultSharePrice = _minVaultSharePrice;
+    }
+
+    /// @notice Sets the max amount of mature positions to close at a time.
+    /// @param _positionClosureLimit Max amount of mature positions to close at
+    ///        a time.
+    function setPositionClosureLimit(
+        uint256 _positionClosureLimit
+    ) external onlyKeepers {
+        _tendConfig.positionClosureLimit = _positionClosureLimit;
+    }
+
+    /// @notice Sets the extra data to pass to hyperdrive when opening/closing
+    ///         longs.
+    /// @param _extraData Extra data to pass to hyperdrive when opening/closing
+    ///         longs.
+    function setExtraData(bytes memory _extraData) external onlyKeepers {
+        _tendConfig.extraData = _extraData;
+    }
+
+    // ╭───────────────────────────────────────────────────────────────────────╮
+    // │                                 VIEWS                                 │
+    // ╰───────────────────────────────────────────────────────────────────────╯
+
+    /// @notice Weighted average maturity timestamp of the portfolio.
+    /// @return Weighted average maturity timestamp of the portfolio.
+    function avgMaturityTime() external view returns (uint128) {
+        return _portfolio.avgMaturityTime;
+    }
+
+    /// @notice Calculates the present portfolio value using the total amount of
+    ///      bonds and the weighted average maturity of all positions.
+    /// @return value The present portfolio value.
+    function calculateTotalAssets() public view returns (uint256 value) {
+        value = ERC20(asset).balanceOf(address(this));
+        if (_portfolio.totalBonds != 0) {
+            // NOTE: The maturity time is rounded to the next checkpoint to
+            //       underestimate the portfolio value.
+            // value += IHyperdrive(hyperdrive).previewCloseLong(
+            //     asBase,
+            //     _poolConfig,
+            //     IEverlongStrategy.Position({
+            //         maturityTime: IHyperdrive(hyperdrive)
+            //             .getCheckpointIdUp(_portfolio.avgMaturityTime)
+            //             .toUint128(),
+            //         bondAmount: _portfolio.totalBonds
+            //     }),
+            //     ""
+            // );
+            for (uint256 i = 0; i < _portfolio.positionCount(); i++) {
+                IEverlongStrategy.Position memory p = _portfolio.at(i);
+                value += IHyperdrive(hyperdrive).previewCloseLong(
+                    asBase,
+                    _poolConfig,
+                    p,
+                    ""
+                );
+            }
```

Comment:
Why are we looping? This seems it would cause issues if we had good usage in an Everlong pool that targeted a 6 month or 1 year Hyperdrive instance. We'd have to do the calculation 180+ times

Reply:
Had put this in when investigating the profit issues yesterday (in case something weird was going on with avg maturity), it wasn't the cause, will remove